src/main/java/com/zendesk/maxwell/schema/MysqlPositionStore.java

```diff
@@ -115,30 +119,15 @@ public synchronized void set(BinlogPosition p) {
 			position.set(p);
 	}
 
-	public void setSync(BinlogPosition p) throws SQLException {
-		LOGGER.debug("syncing binlog position: " + p);
-		set(p);
-		while ( true ) {
-			thread.interrupt();
-			BinlogPosition s = storedPosition.get();
-			if ( p.newerThan(s) ) {
-				try { Thread.sleep(1); } catch (InterruptedException e) { }
-				if ( exception != null )
-					throw(exception);
-			} else {
-				break;
-			}
-		}
-	}
-
 	public BinlogPosition get() throws SQLException {
 		BinlogPosition p = position.get();
 		if ( p != null )
 			return p;
 
 		try ( Connection c = getConnection() ) {
-			PreparedStatement s = c.prepareStatement("SELECT * from `positions` where server_id = ?");
+			PreparedStatement s = c.prepareStatement("SELECT * from `positions` where server_id = ? and client_id = ?");
```

Comment:
one maxwell per *cluster*.  And they'll have to be configured with separate client ids.